packages/web-app-text-editor/src/App.vue

```diff
@@ -1,13 +1,15 @@
 <template>
   <main id="text-editor" class="oc-px-l oc-py-m oc-height-1-1">
-    <app-bar
-      :current-file-context="currentFileContext"
-      :is-loading="isLoading"
-      :is-dirty="isDirty"
-      :is-read-only="isReadOnly"
-      @closeApp="closeApp"
-      @save="save"
-    />
+    <div v-if="isLoading" class="oc-text-center">
+      <oc-spinner :aria-label="$gettext('Loading editor content')" />
+    </div>
+    <app-top-bar v-else :resource="resource" @close="closeApp">
+      <template #right>
+        <oc-button id="text-editor-controls-save" :disabled="isReadOnly || !isDirty" @click="save">
```

Comment:
Would need a label since the only content is an icon (from a screenreader/a11y perspective)?